scripts/run-android.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -m # needed to access jobs
+
+GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
+
+# generates android debug build when we pass the debug flag
+"${GIT_ROOT}"/scripts/build-android.sh
+
+# installs the resulting apk on running emulator or android device
+adb install ./result/app-debug.apk
+
+# start metro and go to background
+nohup "${GIT_ROOT}"/scripts/start-react-native.sh 2>&1 | tee metro-server-logs.log &
+
+echo "Waiting for Metro server to be ready..."
+while ! grep -q "Welcome to Metro" metro-server-logs.log; do
+  echo -n "."
+  sleep 1
+done
```

Comment:
No timeout? I guess it would be a major edge case considering we just started it above.

Reply:
But what if metro starts and then fails? We would wait forever. Shouldn't we timeout or detect metro failure and stop this?

Reply:
Hmm fair enough

Reply:
There is a script we use to wait for `pod install`:
https://github.com/status-im/status-mobile/blob/e86faa07670977e7329193b1d6b4f4797144ecfc/nix/mobile/ios/shells/cocoapods.nix#L17-L18
Maybe you can use it.

Reply:
addressed here : https://github.com/status-im/status-mobile/pull/18784/commits/1920a92d166733e43f97c04a41ae81bad1bd8926